kcidb/orm/__init__.py

```diff
@@ -20,1270 +20,13 @@
-                                f"at position {pos}: {string[pos:]!r}")
-            relation, obj_type_expr, spec, match_spec = \
-                match.group("relation", "type", "spec", "match")
-            obj_id_set, obj_id_set_list = Pattern._parse_spec(
-                spec, obj_id_set_list
-            )
-            try:
-                base_set = Pattern._expand(
-                    schema, base_set, match_set, relation == ">",
-                    obj_type_expr, obj_id_set, match_spec
-                )
-            except Exception as exc:
-                raise Exception(
-                    f"Failed expanding pattern specification "
-                    f"at position {pos}: {string[pos:]!r}"
-                ) from exc
-            pos = match.end()
-        if obj_id_set_list:
-            raise Exception(
-                f"Too many ID sets specified for pattern {string!r}"
-            )
-        return match_set
-
-    @staticmethod
-    def from_io(io_data, schema=None, max_objs=0):
-        """
-        Create a pattern set matching all objects in the supplied I/O data.
-
-        Args:
-            io_data:    The I/O data to create the pattern list from.
-                        Must adhere to the current schema version.
-            schema:     An object type schema to use, or None to use
-                        kcidb.orm.SCHEMA.
-            max_objs:   Maximum number of object IDs to put into each
-                        created pattern (a positive integer).
-                        Zero for no limit.
-
-        Returns:
-            A set of Pattern objects matching the objects in the supplied I/O
-            data.
-        """
-        assert io.SCHEMA.is_compatible_exactly(io_data)
-        assert LIGHT_ASSERTS or io.SCHEMA.is_valid_exactly(io_data)
-        assert schema is None or isinstance(schema, Schema)
-        assert isinstance(max_objs, int) and max_objs >= 0
-
-        if schema is None:
-            schema = SCHEMA
-        # Assert all I/O object lists are represented in the OO schema
-        assert set(schema.types) >= \
-            set(k[:-1] for k in io.SCHEMA.graph if k), \
-            "Specified OO types are not a superset of I/O types"
-        pattern_set = set()
-        for obj_list_name in io.SCHEMA.graph:
-            if not obj_list_name:
-                continue
-            assert obj_list_name.endswith("s")
-            obj_list = io_data.get(obj_list_name, [])
-            if not obj_list:
-                continue
-            obj_type = schema.types[obj_list_name[:-1]]
-            id_fields = obj_type.id_fields
-            for obj_list_slice in kcidb.misc.isliced(obj_list, max_objs):
-                pattern_set.add(
-                    Pattern(None, True, obj_type, {
-                        tuple(o[id_field] for id_field in id_fields)
-                        for o in obj_list_slice
-                    })
-                )
-        return pattern_set
-
```

Comment:
Most (if not all of these) need to stay in this module. Please only move the argparse stuff. Post questions in the issue, this PR, or on the #outreachy channel, if you have a difficulty deciding what should stay and what should move, or disentangling them. In general the approach I describe for the `db` module should be used.

Reply:
I thought we will also need to move dependent functions also as the argparse functions were throwing errors. Should I import those functions that argparse_ ones require into the argparse.py ?

Reply:
The way you normally do this is by parametrizing the functions/classes being extracted from the original module, and then supplying whatever they need to them when calling/using them. Similarly to what I described for the `db` module. Let's chat on the Slack channel, or maybe have a quick call tomorrow, if this is not clear enough :slightly_smiling_face: